docs/auth/jwt_auth.rst

```diff
@@ -0,0 +1,241 @@
+            payload["aud"] = audience
+        # 'sub' is the primary contents of a JWT, so I user "user_id" here
+        payload['sub'] = user_id
+        # available permissions associated to this user or JWT, not required here
+        # but helpful to explain how they work
+        payload['permissions'] = permissions
+        # again an extension of the token, we will explain further on why only user_id
+        # is required, but passing back to the frontend can be very useful.
+        payload['user_name'] = user_name
+        token = jwt.encode(payload, key, algorithm=JWT_ALGO, json_encoder=None)
+        if not isinstance(token, str):  # Python3 unicode madness
+            token = token.decode("ascii")
+        return token
+
+Now the token should look something like::
+
+    eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjY1MDM5Ny4wLCJzdWIiOjk5OSwicGVybWl
+    zc2lvbnMiOlsidmlldyIsImVkaXQiXSwidXNlcl9uYW1lIjoiam9lIn0.dTCQxCorZtzIExeUtxB49_J
+    AKljS9M8FZEhBvT_JPudzGuOyTPGYpTaxgaYWEjvnMG1m_kKvASJcn77Klgb9lQ
+
+Now, we can work onto the next phase, intercepting this token in the security policy
+and evaluating it. It is important to note, that all data inside a JWT is available 
+for anyone to view it, all the secret does is verify the token, so NEVER send information
+inside a JWT that you don't mind getting out.
+
+
+Evaluating with pyramid 
+#######################
+
+So, now we have send the JWT token to our front-end application, we can presume that
+it will get sent back to the pyramid application (`In the header
+<https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.authorization>`_
+) for evaluation to access protected views. So we will create a security policy to handle this.
+
+Lets assume we have a protected view::
+
+    @view_config(route_name='view_basket', renderer='json', permission="view")
+    def view_basket(request):
+
+Only a JWT token that has the permission "view" *should* be able to access
+this view, so how does this work?
+
+So, before we delve into our Security Policy, we need to be able to decode our
+JWT token, here is an example of how this *could* be done (you can read more
+on the `PyJWT Docs <https://pyjwt.readthedocs.io/en/stable/>`_ )::
+
+    def decode_jwt_token(token, secret):
+        ''' Function to decode our JWT token
+        '''
+        try:
+            decode_token = jwt.decode(
+                token,
+                secret,
+                algorithms=JWT_ALGO,
+                leeway=0,
+                audience=None,
+            )
+            return {
+                'user_id': decode_token['sub'],
+                'user_name': decode_token['user_name'],
+                'permissions': decode_token['permissions']
+            }
+        except jwt.exceptions.InvalidTokenError as e:
+            # Invalid token detected
+            return None
+
+Nice and simple, if our token is valid and not expired, we will return 
```

Comment:
Avoid "simple" ("easy" or "just do ___" are other phrases), as this can frustrate a reader who is just learning auth stuff, which is *hard* for them.

```suggestion
If our token is valid and not expired, we will return 
```